receiver/splunkhecreceiver/receiver.go

```diff
@@ -295,18 +378,44 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, err)
 		return
 	}
+
 	consumerErr := r.logsConsumer.ConsumeLogs(ctx, ld)
 
 	_ = bodyReader.Close()
 
 	if consumerErr != nil {
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, consumerErr)
 	} else {
-		r.writeSuccessResponse(ctx, resp, ld.LogRecordCount())
+		if len(partitionID) > 0 && r.ackExt != nil {
+			r.writeSuccessResponseWithAck(ctx, resp, ld.LogRecordCount(), partitionID)
+		} else {
+			r.writeSuccessResponseWithoutAck(ctx, resp, ld.LogRecordCount())
+		}
 		r.obsrecv.EndLogsOp(ctx, metadata.Type.String(), slLen, nil)
 	}
 }
 
+func (r *splunkReceiver) validateAndExtractChannelHeader(req *http.Request) (string, error) {
```

Comment:
Nit: Maybe make validation and extraction two separate methods? Or just call it extract? The fact that it returns an error makes it pretty clear there's validation going on. Extract could call the validate method internally if you split it up. Not a big deal for now though.

Reply:
refactored in to two methods